### PR TITLE
Add strange-forms examples and GitHub pages

### PR DESCRIPTION
**Pull request by devinivy**
The intention of this work is to document clearly how strange-forms works and how one might use it in various situations, through example.  The provided examples are:

 - Uncontrolled - an uncontrolled form.
 - Controlled - a controlled form.
 - Complex Value - an uncontrolled form with a complex form value for WYSIWYG input state.
 - Nested - composing the controlled form from the Controlled example into a new controlled form with an additional field.
 - Constraints - applying constraints on user input.
 - List - composing arbitrarily many copies of the controlled form from the Controlled example into a list of independent forms, acting as a single controlled form.

These are deployed to GitHub pages here: https://bigroomstudios.github.io/strange-forms.  Each page also contains a short explainer, mentioning general pitfalls that may arise when implementing this example.

In order to illustrate the inner-workings of strange-forms, there is also a concise reference implementation using hooks, which is used to re-implement versions of the Controlled and Uncontrolled examples.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ - Uncontrolled w/ hooks [[form](./src/components/Forms/UncontrolledHook.js)] - may be swapped into any example using UncontrolledForm just by altering the `require()` location.
+
+## Requirements
+* node `12.x.x`
+
+## Getting Started
+Click the green "Use this template" button above or install manually using git:
+
+```bash
+$ git clone --depth=1 --origin=strangeluv --branch=strangeluv git@github.com:bigroomstudios/strangeluv.git my-project
```
Comment (bmleight) [inline examples/README.md:29]:
```suggestion
$ git clone git@github.com:BigRoomStudios/strange-forms.git
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+$ cd my-project
```
Comment (bmleight) [inline examples/README.md:30]:
```suggestion
$ cd strange-forms/examples
```

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+$ git checkout --orphan master  # New branch without history
```
Comment (bmleight) [inline examples/README.md:31]:
```suggestion
```

File: examples/src/routes/forms/components/ComplexValuePage.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        />
+    </FormValuesLayout>;
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="Complex Value"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/ComplexValue.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/ComplexValuePage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/ComplexValuePage.js:29]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/ComplexValue.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/ComplexValuePage.js"
```

File: examples/src/routes/forms/components/ConstraintsPage.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        />
+    </FormValuesLayout>;
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="Constraints"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/Constraints.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/ConstraintsPage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/ConstraintsPage.js:27]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/Constraints.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/ConstraintsPage.js"
```

File: examples/src/routes/forms/components/ControlledPage.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        spam: false
+    }
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="Controlled"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/Controlled.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/ControlledPage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/ControlledPage.js:75]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/Controlled.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/ControlledPage.js"
```

File: examples/src/routes/forms/components/ListPage.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        </ListForm>
+    </FormValuesLayout>;
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="List"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/List.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/ListPage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/ListPage.js:38]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/List.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/ListPage.js"
```

File: examples/src/routes/forms/components/NestedPage.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        </NestedForm>
+    </FormValuesLayout>;
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="Nested"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/Nested.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/NestedPage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/NestedPage.js:38]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/Nested.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/NestedPage.js"
```

File: examples/src/routes/forms/components/UncontrolledPage.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        spam: false
+    }
+};
+
+internals.Explainer = () => (
+
+    <FormExplainer
+        header="Uncontrolled"
+        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/components/Forms/Uncontrolled.js"
+        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/examples/examples/src/routes/forms/components/UncontrolledPage.js"
```
Comment (devinivy) [inline examples/src/routes/forms/components/UncontrolledPage.js:64]:
```suggestion
        formUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/components/Forms/Uncontrolled.js"
        pageUrl="https://github.com/BigRoomStudios/strange-forms/blob/master/examples/src/routes/forms/components/UncontrolledPage.js"
```

**Review by bmleight — APPROVED** [review]
Sorry for the delayed review (and for submitting this right after you merge it in...)!  I started this earlier in the week, and then it got lost in my sea of tabs.  I just made a few suggestions on the install instructions for the examples.  Thanks for the great examples, it was nice to checkout some strangeluv code again.

**Pull request merged**